jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -345,10 +352,12 @@ public void continue100(int available) throws IOException
         }
     }
 
-    public void recycle()
+    private void recycle()
     {
         _written = 0;
         _oldIdleTimeout = 0;
+        _httpInput.recycle();
+        _callback = null;
```

Comment:
Can we clear out things like `_servletContextRequest` and `_servletContextApi` as well.